core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -75,15 +75,14 @@ void HIP::impl_initialize(InitializationSettings const& settings) {
   }
 #endif
 #ifdef KOKKOS_ARCH_AMD_GFX942_APU
-  if (!Kokkos::Impl::xnack_enabled()) {
+  if (!Kokkos::Impl::xnack_environment_enabled()) {
```

Comment:
```suggestion
  if (!Kokkos::Impl::xnack_hmm_found()) {
```
would match that the warning is talking about setting `HSA_XNACK=1`.